### PR TITLE
Use dracut $hookdir instead of hardcoding

### PR DESCRIPTION
**Pull request by lnussel**
Dracut hookdir will move. See https://bugzilla.suse.com/show_bug.cgi?id=1218068

Still testing it, posting already so you know what's coming

### CODE REVIEW

File: combustion [hunk 1]
```diff
@@ -185,12 +188,16 @@ mount --make-private /sysroot
...
-if [ -e /lib/dracut/hooks/pre-pivot/50-selinux-microos-relabel.sh ]; then
-	NEWROOT=/sysroot bash -c '. /lib/dracut-lib.sh; . /lib/dracut/hooks/pre-pivot/50-selinux-microos-relabel.sh'
-elif [ -e /sysroot/.autorelabel ] || [ -e /sysroot/etc/selinux/.autorelabel ]; then
-	echo "Relabel (probably) needed, but selinux-microos-relabel not found."
-	exit 1
-fi
+(
+	set +xu
+	. /lib/dracut-lib.sh
+	if [ -e "$hookdir"/pre-pivot/50-selinux-microos-relabel.sh ]; then
```
Comment (Vogtinator) [inline combustion:194]: I prefer the

```suggestion
	if [ -e "${hookdir}/pre-pivot/50-selinux-microos-relabel.sh" ]; then
```

style

File: combustion [hunk 2]
```diff
@@ -185,12 +188,16 @@ mount --make-private /sysroot
...
+		NEWROOT=/sysroot bash -c '. /lib/dracut-lib.sh; . $hookdir/pre-pivot/50-selinux-microos-relabel.sh' || true
```
Comment (Vogtinator) [inline combustion:195]: `|| true`?

File: combustion [hunk 3]
```diff
@@ -100,15 +101,17 @@ if [ "${1-}" = "--prepare" ]; then
...
 				# Re-trigger generation of network rules and apply them
 				udevadm control --reload
 				udevadm trigger --subsystem-match net --action add
-			else
+			elif [ -e "\$hookdir"/cmdline/99-nm-config.sh ]; then
 				# NetworkManager
-				. /lib/dracut/hooks/cmdline/99-nm-config.sh
+				. "\$hookdir"/cmdline/99-nm-config.sh
+			else
+				echo "ERROR: unknown network framework" >&2
```
Comment (Vogtinator) [inline combustion:114]: If it's an error it should `exit 1`

File: combustion [hunk 4]
```diff
@@ -100,15 +101,17 @@ if [ "${1-}" = "--prepare" ]; then
 			# Set rd.neednet if not already done and reevaluate it (module-specific)
 			getargbool 0 'rd.neednet' && exit 0
 			echo rd.neednet=1 > /etc/cmdline.d/40-combustion-neednet.conf
-			if [ -e /lib/dracut/hooks/pre-udev/60-net-genrules.sh ]; then
+			if [ -e "\$hookdir"/pre-udev/60-net-genrules.sh ]; then
```
Comment (Vogtinator) [inline combustion:104]: If the heredoc is converted to `<<"EOF"` or the sh call to a subshell this won't have to be escaped

**Review by Vogtinator — COMMENTED** [review]
Looks mostly good already

File: combustion (hunk 3 above)
Comment (lnussel) [reply combustion:113]: was hoping to get away with incomplete networkd support for now

**Review by Vogtinator — APPROVED** [review]

**Pull request merged**